src/debug.c

```diff
@@ -1136,22 +1136,26 @@ void _serverPanic(const char *file, int line, const char *msg, ...) {
 #ifdef HAVE_BACKTRACE
         logStackTrace(NULL, 1);
 #endif
-        printCrashReport();
+        if (new_report) printCrashReport();
     }
 
     // remove the signal handler so on abort() we will output the crash report.
     removeSigSegvHandlers();
     bugReportEnd(0, 0);
 }
 
-void bugReportStart(void) {
+/* Start a bug report, returning 1 if this is the first time this function was called, 0 otherwise. */
+int bugReportStart(void) {
+    int new_report = 0;
     pthread_mutex_lock(&bug_report_start_mutex);
     if (bug_report_start == 0) {
         serverLogRaw(LL_WARNING|LL_RAW,
         "\n\n=== REDIS BUG REPORT START: Cut & paste starting from here ===\n");
         bug_report_start = 1;
+        new_report = 1;
```

Comment:
Why do we need both `new_report` and  `bug_report_start`? They are always aligned.
Also, I suggest eliminating the lock and making `bug_report_start` atomic

Reply:
bug_report_start is a global, so it's always 1 at the end of this function regardless if this function was the one that set it, not when this was the first report.